src/components/calculator.js

```diff
@@ -1,41 +1,59 @@
 import React from 'react';
 import './calculator.css';
+import calculate from '../logic/calculate';
+import '../logic/operate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: 0,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((previousState) => calculate(previousState, e.target.innerText));
   }
 
   render() {
+    let val = 0;
+    const { total, next } = this.state;
+    if (next) {
+      val = next;
+    } else if (total) {
+      val = total;
+    }
+
     return (
       <div className="wrapper">
 
-        <input type="text" className="operator display" id="calc-screen" placeholder="0" disabled />
+        <input type="text" className="operator display" id="calc-screen" placeholder="0" value={val} />
 
-        <button type="button" className="operator all-clear" name="AC">AC</button>
-        <button type="button" className="operator Plus-minus operations" name="+/-">+/-</button>
-        <button type="button" className="operator" name="%">%</button>
-        <button type="button" className="operator operation" name="+">+</button>
+        <button type="button" className="operator all-clear" name="AC" onClick={this.handleClick}>AC</button>
+        <button type="button" className="operator Plus-minus operations" name="+/-" onClick={this.handleClick}>+/-</button>
+        <button type="button" className="operator" name="%" onClick={this.handleClick}>%</button>
+        <button type="button" className="operator operation" name="+" onClick={this.handleClick}>+</button>
```

Comment:
[ Optional Changes ]

- [ ] I believe, according to the design, it is supposed to be the division sign that should be here and not the addition sign. You must have interchanged them.